src/main/java/utils/Parser.java

```diff
@@ -197,37 +200,54 @@ private Command prepareDeleteDishCommand(String[] userInputSplit) {
         return new DeleteDishCommand(indexToRemove);
     }
 
-    private Command prepareViewDishCommand() {
+    private Command prepareViewDishCommand(String userInputNoCommand) {
+        try {
+            if (!userInputNoCommand.isBlank()) {
+                throw new DinerDirectorException(Messages.ERROR_COMMAND_INVALID);
+            }
+        } catch (DinerDirectorException e) {
+            return new IncorrectCommand();
+        }
         return new ViewDishCommand();
     }
 
-    private Command prepareAddDishCommand() {
+    private Command prepareAddDishCommand(String userInputNoCommand) {
         //call for subsequent inputs:
         // name of dish?
         // price of dish?
         // list of ingredients, maybe indicate number of ingredients followed by listing them
 
+        try {
+            if (!userInputNoCommand.isBlank()) {
+                throw new DinerDirectorException(Messages.ERROR_COMMAND_INVALID);
+            }
+        } catch (DinerDirectorException e) {
+            return new IncorrectCommand();
+        }
+
         String name;
         int price;
         ArrayList<String> ingredients;
 
         try {
-            System.out.println("Name of Dish?");
             Scanner userInput = new Scanner(System.in);
+            System.out.println("Name of Dish?");
             name = userInput.nextLine();
             if (name.isBlank()) {
                 throw new DinerDirectorException(Messages.MESSAGE_BLANK_DISH_NAME_COMMAND);
             }
 
             System.out.println("Price of Dish? (In cents)");
-            userInput = new Scanner(System.in);
-            price = Integer.parseInt(userInput.nextLine());
+            String priceInString = userInput.nextLine();
+            price = Integer.parseInt(priceInString);
             if (price < 0) {
                 throw new DinerDirectorException(Messages.MESSAGE_NEGATIVE_PRICE_COMMAND);
             }
 
             System.out.println("List of ingredients? (Separate it by spaces)");
-            userInput = new Scanner(System.in);
+            if (!userInput.hasNext()) {
+                throw new DinerDirectorException(Messages.MESSAGE_INGREDIENT_LIST_CANNOT_BE_EMPTY);
+            }
             String[] userInputSplit = userInput.nextLine().split(" ");
```

Comment:
are you sure all ingredients can be split with " ", like baking powder or smt, maybe ask them to split with something else. like i/name1 i/name2 i/name3 than split with " i/" will be better?

also i think u need to check with Darren regarding the help guide cos urs is diff from our user guide commands.